builtin/mainmenu/tab_settings.lua

```diff
@@ -15,497 +15,504 @@
 --with this program; if not, write to the Free Software Foundation, Inc.,
 --51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
---------------------------------------------------------------------------------
+local FILENAME = "settingtypes.txt"
 
-local leaves_style_labels = {
-	fgettext("Opaque Leaves"),
-	fgettext("Simple Leaves"),
-	fgettext("Fancy Leaves")
-}
+local function parse_setting_line(settings, line)
+	-- empty lines
+	if line:match("^[%s]*$") then
+		-- clear current_comment so only comments directly above a setting are bound to it
+		settings.current_comment = ""
+		return
+	end
 
-local leaves_style = {
-	{leaves_style_labels[1]..","..leaves_style_labels[2]..","..leaves_style_labels[3]},
-	{"opaque", "simple", "fancy"},
-}
+	-- category
+	local category = line:match("^%[([^%]]+)%]$")
+	if category then
+		local level = 0
+		local index = 1
+		while category:sub(index, index) == "*" do
+			level = level + 1
+			index = index + 1
+		end
+		category = category:sub(index, -1)
+		table.insert(settings, {
+			name = category,
+			level = level,
+			type = "category",
+		})
+		return
+	end
+
+	-- comment
+	local comment = line:match("^#[%s]*(.*)$")
+	if comment then
+		if settings.current_comment == "" then
+			settings.current_comment = comment
+		else
+			settings.current_comment = settings.current_comment .. "\n" .. comment
+		end
+		return
+	end
 
-local dd_filter_labels = {
-	fgettext("No Filter"),
-	fgettext("Bilinear Filter"),
-	fgettext("Trilinear Filter")
-}
+	-- settings
+	local first_part, name, readable_name, type =
+			line:match("^(([%w%._-]+)[%s]+%(([^%)]*)%)[%s]+([%w_]+)[%s]*)")
```

Comment:
* No, since the type dependent parameters follow
* Is this really an issue here?